cigaradvisor/blocks/search-results/search-results.js

```diff
@@ -93,21 +14,23 @@ function filterData(fullTerm, data) {
  * @param {number} articlesCount - The total count of articles. This is needed for pagination.
  * @returns {Promise<void>} - A promise that resolves when the page is rendered.
  */
-async function processSearchResults(results, allArticles, wrapper, limit, articlesCount) {
-  const articles = [];
-  results.forEach((post) => {
-    const filteredArticles = allArticles.filter((obj) => obj.path === getRelativePath(post.path));
-    articles.push(filteredArticles[0]);
-  });
+async function processSearchResults(articles, wrapper, limit, articlesCount) {
   await renderPage(wrapper, articles, limit, articlesCount);
   const loadingImageContainer = document.querySelector('.loading-image-container');
   const articleListWrapper = document.querySelector('.article-list-wrapper');
+  loadingImageContainer.style.transition = 'opacity 2s';
+  loadingImageContainer.style.opacity = 0.25;
   articleListWrapper.style.transition = 'opacity 2s';
-  articleListWrapper.style.opacity = 1;
+  articleListWrapper.style.opacity = 0.5;
   articleListWrapper.style.display = 'block';
-  loadingImageContainer.style.transition = 'opacity 2s';
-  loadingImageContainer.style.opacity = 0;
-  loadingImageContainer.style.display = 'none';
+
+  setTimeout(
```

Comment:
No need to timeout here. as long as all the articles are rendered and added to the dom, we can just apply the css.

Reply:
Fixed. But loading of 8 teasers is almost instantaneous and the fade-out effect won't be visible without timer.